components/TaskItem.tsx

```diff
@@ -4,7 +4,7 @@ import { AntDesign } from "@expo/vector-icons";
 import { useState } from "react";
 import { useSQLiteContext } from "expo-sqlite";
 
-export default function TaskItem({ task }) {
+export default function TaskItem({ task, onDelete }) {
   const [done, setDone] = useState(task.isDone);
```

Comment:
In this case, and application, consolidate the state into a context provider which should be at the root or let the parent handle all the states. Since the components using them are closely related.

Define a ContextProvider, that will expose these props:

```TSX

type Task {
  id: string; // or number
  title: string;
  category: string;
  // ... other properties
}

type TaskManagerContextType = {
  // central state
  tasks: Task[],

  // define your CRUD operation
  addTask: (task: Task) => Promise<boolean>;
  removeTask: (taskId: string) => Promise<boolean>;
  updateTask: (task: Task) => Promise<boolean>;
  deleteTask: (task: Task) => Promise<boolean>;
}

// Note: Promise is returned since database operations are async in nature, it will return true if success or false if fails. This state can be useful to display loading indicator while promise is pending and checkmark on succes or warning message on failure.
```

The idea is to control the related state from one point, which is the context-provider in this case. 

As a general rule of thump:

- Place state as close to where it's used as possible, lifting it only to common ancestor components when necessary to avoid duplication across siblings.
- Ideally there should be always be a single source of truth. i.e. on updating a task, if you have state defined twice - once in local component and once in parent component, you are duplicating the state.
- Minimise state-fullness of a component
  - Notice that more state a component defines less re-usable it is.
  - Eg. TaskItem should not define any internal state and rely on it's component props. This will enable consistent and predictable behaviour.
  - Prop-drilling is an issue sometimes, which can be overcome by re-arranging how state is managed or flattening the childrens into siblings or use a context provider.
- Propagate State Up as Needed: When multiple components need access to the same state, move the state up to their closest common ancestor. This method, known as "lifting state up," helps avoid prop drilling and keeps the component hierarchy clean.


The way you define your application state and data-structures you use (array, instead of set or Record of key and object value, etc) has large impact on your overall application design and its' scalability. 

A good state and design would be highly intuitive and predicatable. With a single glance you could deduce the inner workings. It's more easier to identify a good design than a bad one.